### PR TITLE
Update/add remaining terraform plan commands

### PR DESCRIPTION
**Pull request by keithcampbelljr**
Added most of the plan configuration arguments
Added tons of new tests
Refactor plan method to use a command mapping to ensure we don't have tons of if statements

### CODE REVIEW

File: src/builders/terraform_command_builder.py [hunk 1]
```diff
@@ -7,15 +9,50 @@ def __init__(self):
     def init(self, backend_config=None):
         self.command_string += " init"
         if backend_config:
-            self.command_string += f' -reconfigure -backend-config="{backend_config}"'
+            self.command_string += f' -reconfigure -backend-config={backend_config}'
         return self
 
-    def plan(self, var_file_name):
+    def plan(self, 
+            var_file_name: str = None,
```
Comment (devanbenz) [inline src/builders/terraform_command_builder.py:16]: Is it possible to implement some sort of object/struct (whatever they are called in python) for these variables? This is a ton of params for a function signature.

**Review by devanbenz — CHANGES_REQUESTED** [review]

File: src/builders/terraform_command_builder.py (hunk 1 above)
Comment (devanbenz) [reply src/builders/terraform_command_builder.py:16]: If you ever use this object else where its good to have one place where its defined.

Comment (keithcampbelljr) [reply src/builders/terraform_command_builder.py:16]: Yeh, but my counter is a lot of these are specific to a terraform plan and wouldn’t be used elsewhere.

Comment (devanbenz) [reply src/builders/terraform_command_builder.py:16]: Gotcha - so the only place where you would define the plan is in this function?

**Review by devanbenz — APPROVED** [review]

**Pull request merged**